.github/workflows/maven-automatic.yml

```diff
@@ -0,0 +1,61 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Automatic Package
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    name: "Build Job"
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+        server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+        settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+    - name: Build with Maven
+      run: mvn -Drevision=1145.vb_cf6cf6ed960 -B package --file pom.xml
+    
+    - run: mkdir staging && cp target/*.hpi staging
+    - uses: actions/upload-artifact@v3
+      with:
+        name: Package
+        path: staging/*
+        if-no-files-found: error
+        
+  deploy:
+    runs-on: ubuntu-latest
+    name: "Deploy Job"
+    needs: build
+      
+    steps:
+    - name: Download package
+      uses: actions/download-artifact@v3
+      with:
+        name: Package
+    
+    - name: Create release
+      uses: "marvinpinto/action-automatic-releases@latest"
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        automatic_release_tag: 'latest'
+        prerelease: ${{ env.GITHUB_EVENT_NAME == 'pull_request' }}
+        draft: false
+        title: "Latest Release Build"
+        files: |
+          *.hpi
```

Comment:
We actually wouldn't want to auto create a release on every commit. Instead we'd just want to run the first steps (build & package)